backend/pkg/sqlretry/deadlock.go

```diff
@@ -29,8 +30,8 @@ func isMysqlDeadlockError(err error) bool {
 	if err == nil {
 		return false
 	}
-	mysqlErr, ok := err.(*mysql.MySQLError)
-	if ok {
+	var mysqlErr *mysql.MySQLError
+	if errors.As(err, &mysqlErr) {
```

Comment:
we know that this already works as we are seeing proper deadlock detection errors for mysql users in our logs, but this is an extra safeguard if these errors ever become wrapped.